File: articles/expected-value-and-expected-utility/r/run_all_expected_utility_workflows.R
# run_all_expected_utility_workflows.R
# Run all R workflows for Expected Value and Expected Utility.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "expected_value_expected_utility_workflow.R",
  "expected_value_profiles.R",
  "utility_curve_sensitivity.R",
  "certainty_equivalent_report.R",
  "risk_premium_profiles.R",
  "probability_assumption_audit.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R expected-utility workflows completed.")

File: articles/expected-value-and-expected-utility/python/run_all_expected_utility_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Expected Value and Expected Utility."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "expected_value_calculator.py",
    "utility_function_profiles.py",
    "certainty_equivalent_calculator.py",
    "risk_premium_analysis.py",
    "risk_aversion_sensitivity.py",
    "probability_quality_audit.py",
    "decision_record_exporter.py",
    "expected_value_expected_utility_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python expected-utility workflows completed.")


if __name__ == "__main__":
    main()
